Sources/Common/Core/DataArray/index.js

```diff
@@ -281,6 +281,22 @@ function vtkDataArray(publicAPI, model) {
     return model.rangeTuple;
   };
 
+  publicAPI.getRanges = () => {
```

Comment:
Look at `getRange()`: for multi-components, model.ranges has the size of numberOfComponents + 1 (where the last item is the range across all components):
```
...
if (model.numberOfComponents > 1) {
  const [min, max] = publicAPI.getRange(-1);
  ...
  ranges.push(range);
}
return ranges;
};`

Reply:
I think an option to not enforce range computation might be useful because calling `getRanges()` might end up being more costly than needed (e.g. let's say you are only using with the range of the first component (or the overall range) of a multi-component array, calling `getRanges()` would end up computing the range of all the components):
```
publicAPI.getRanges => (computeRanges=true) {
  if (!computeRanges) {
     return DeepCopy(model.ranges);
  }
  ...
}
```

Reply:
I can add this. I see that `structuredClone` is used elsewhere.  Should I use `structuredClone` (https://caniuse.com/?search=structuredClone) or is there a preferred method for `deepCopy`?

Reply:
`structuredClone` is fine by me.